### PR TITLE
Add helm_push_registry

### PR DESCRIPTION
**Pull request by iDiogenes**
* Add the ability to push a helm package to an oci registry

### CODE REVIEW

**Comment by iDiogenes** [comment]
Opening this PR based on issue #28. While the issue talked about using ChartMuseum, I opted to go the route of pushing to an oci registry. Like with rules_oci, this commit leverages helm's internals for authentication when pushing to a registry. 

Also, something to consider, which I think might be a better approach, rather than using a new rule `helm_push_registry`, I could update this to extend `helm_push` to push the charts to an oci registry if a `registry_url` is provided. Personally, I think this is better approach, but I didn't want to go this route as we had not discussed it. Please let me know. 

Cheers!

File: helm/private/registrar/helm_registry.sh.template [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+eval {helm} push {chart} {registry_url}
```
Comment (abrisco) [inline helm/private/registrar/helm_registry.sh.template:5]: should be able to exec, right?
```suggestion
exec {helm} push {chart} {registry_url}
```

File: helm/private/registrar/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        ":windows": ":helm_registry.bat.template",
+        "//conditions:default": ":helm_registry.sh.template",
+    }),
+)
+
+filegroup(
+    name = "bzl_srcs",
+    srcs = glob(
+        ["**/*.bzl"],
+        allow_empty = True,
```
Comment (abrisco) [inline helm/private/registrar/BUILD.bazel:25]: does this need allow_empty?

**Review by abrisco — CHANGES_REQUESTED** [review]
This seems great, thanks! Just a couple of questions and some complaints from CI left 😄

**Comment by abrisco** [comment]
I applied these changes on https://github.com/abrisco/rules_helm/pull/44, thank you so much for opening this PR!

**Comment by iDiogenes** [comment]
@abrisco - Sorry for the delay in resolving the feedback requests, I had a very busy week and was planing on getting to it this weekend. Thanks for taking care of it!

**Comment by abrisco** [comment]
> @abrisco - Sorry for the delay in resolving the feedback requests, I had a very busy week and was planing on getting to it this weekend. Thanks for taking care of it! 

No worries at all! Someone had asked me about exactly this feature and besides a few nits your change looked perfect! Thought I'd do everyone a favor and rebase it and merge. Thank you again for the change!

**Pull request closed without merging**